src/lib/jsonParseV2/parseEntitiesVector.cpp

```diff
@@ -0,0 +1,222 @@
+          *errorStringP = "forbidden characters in /entities/ vector item /id/";
+          return false;
+        }
+        if (idOpt.value.length() > MAX_ID_LEN)
+        {
+          *errorStringP = "max id length exceeded in /entities/ vector item /id/";
+          return false;
+        }
+        id = idOpt.value;
+      }
+    }
+
+    {
+      Opt<std::string> idPatOpt = getStringOpt(*iter, "idPattern", "/entities/ vector item /idPattern/");
+
+      if (!idPatOpt.ok())
+      {
+        *errorStringP = idPatOpt.error;
+        return false;
+      }
+      else if (idPatOpt.given)
+      {
+        if (idPatOpt.value.empty())
+        {
+          *errorStringP = "/entities/ vector item /idPattern/ is empty";
+          return false;
+        }
+
+        idPattern = idPatOpt.value;
+
+        // FIXME P5: Keep the regex and propagate to sub-cache
+        regex_t re;
+        if (regcomp(&re, idPattern.c_str(), REG_EXTENDED) != 0)
+        {
+          *errorStringP = ERROR_DESC_BAD_REQUEST_INVALID_REGEX_ENTIDPATTERN;
+          return false;
+        }
+        regfree(&re);  // If regcomp fails it frees up itself
+      }
+    }
+
+    {
+      Opt<std::string> typeOpt = getStringOpt(*iter, "type", "/entities/ vector item type");
+
+      if (!typeOpt.ok())
+      {
+        *errorStringP = typeOpt.error;
+        return false;
+      }
+      else if (typeOpt.given)
+      {
+        if (forbiddenIdCharsV2(typeOpt.value.c_str()))
+        {
+          *errorStringP = "forbidden characters in /entities/ vector item /type/";
+          return false;
+        }
+        if (typeOpt.value.length() > MAX_ID_LEN)
+        {
+          *errorStringP = "max type length exceeded in /entities/ vector item /type/";
+          return false;
+        }
+        if (typeOpt.value.empty())
+        {
+          *errorStringP = ERROR_DESC_BAD_REQUEST_EMPTY_ENTTYPE;
+          return false;
+        }
+        type = typeOpt.value;
+      }
+    }
+
+    {
+      Opt<std::string> typePatOpt = getStringOpt(*iter, "typePattern", "/entities/ vector item /typePattern/");
+
+      if (!typePatOpt.ok())
+      {
+        *errorStringP = typePatOpt.error;
+        return false;
+      }
+      else if (typePatOpt.given)
+      {
+        if (typePatOpt.value.empty())
+        {
+          *errorStringP = "/entities/ vector item /typePattern/ is empty";
+          return false;
+        }
+
+        typePattern = typePatOpt.value;
+
+        // FIXME P5: Keep the regex and propagate to sub-cache
+        regex_t re;
+        if (regcomp(&re, typePattern.c_str(), REG_EXTENDED) != 0)
+        {
+          *errorStringP = ERROR_DESC_BAD_REQUEST_INVALID_REGEX_ENTTYPEPATTERN;
+          return false;
+        }
+        regfree(&re);  // If regcomp fails it frees up itself
+      }
+    }
+
+
+    ngsiv2::EntID  eid(id, idPattern, type, typePattern);
+
+    //
+    // FIXME PR: the following commented line gives strange compilation errors in Ubuntu 17.04
```

Comment:
This here is important.
I had to remove this check. Didn't compile ...
Really cryptic message from the C++ compiler, next to impossible to understand.

The check is important, should not be removed!

Reply:
A bit offtopic but, how did you get to this check at the begining? I mean, is parseEntitiesVector.cpp partly based on another previously existing file where this check were?

Reply:
(Another possibility is this file is moved from elsewhere and git failed to show it as moved instead of as new file)

Reply:
I got this problem as I changed some type ... somewhere ...
C++ ... 
Need to fix this though !!!
The check is important!

FIXME PR

Reply:
A type changed, I think ...

Reply:
Just put this comment to not forget about it

Reply:
And 'yes', parseEntitiesVector.cpp may be a new file but all its contents is old ...

Reply:
Dobut: from where did the old content come? I mean, which file.

Reply:
It came from parseSubscription.cpp

Reply:
Fixed in 2ff7688ea15ee6ae5fdc3c4db20dc39d848c0405

I simply removed this check:

`if (std::find(eivP->begin(), eivP->end(), eid) == eivP->end())`
before pushing the `eid`

This check is very cryptic ...
Seems that it is to avoid to push an identical entity twice onto the vector.
No idea how it works.

However, we're parsing the JSON payload here, so if an entity it present twice, it's because the user has given it that way ...

I don't really see how the check is necessary.
If we want the check, instead of silently ignoring the entity, we should return an error.
And the check should be implemented in a way that is easy understandable.

IMO: the check is not necessary, and if we want it in the future, we'll implement this in a better way

Reply:
If this check comes from the subscription code probably it is also unneded there. Could you add a comment in that part (subscription) about it so we could remember to remove it in the future if at the end is not needed also there?

Reply:
Well, the check was moved from the subscription code to this separate file, to be used by both. there is no copy

Reply:
NTC